### ISSUE TITLE
FEATURE REQUEST: Move Material Components' styleUrls to a global stylesheet similar to traditional CSS.

### ISSUE THREAD

**Issue by classicmike** [angular/components#16849]
#### Bug, feature request, or proposal:
I would like to propose the moving of all styles declared in Angular Material's Components' styleUrl property to be moved to a global stylesheet that users import manually through many means such as ```style``` tag, ```@import``` etc...

#### What is the expected behavior?
The expected behavior is that I do NOT want angular material to load any default styles through the ```styleUrls``` property of the component and rather load it via a global stylesheet. 

That way if I am prepared to take responsibility on the consequences of not loading any of those default styles I can do it without having to resort to class overriding or other techniques.

#### What is the current behavior?

Just so you have a perspective, my colleagues are consuming components such as Angular Material's ```MatFormField``` with ```matInput``` directives mainly for the purposes of using the functionality to do things such as showing validation errors, showing hint text of the number of characters remaining inside of a text field etc. However, ```MatFormField``` component injects the following CSS in its style urls:

Below fragment taken from https://github.com/angular/components/blob/master/src/material/form-field/form-field.ts

```
styleUrls: [
    'form-field.css',
    'form-field-fill.css',
    'form-field-input.css',
    'form-field-legacy.css',
    'form-field-outline.css',
    'form-field-standard.css',
  ]
```

#### What are the steps to reproduce?
Simply look at this file and the class ```MatFormField``` in https://github.com/angular/components/blob/master/src/material/form-field/form-field.ts

#### What is the use-case or motivation for changing an existing behavior?
My colleagues wish to use Angular Material mainly for the reason it takes care of most of the javascript functionality required without having to write their own code to do the same thing. 

However, we have our own CSS framework which has stylings exactly to our organisation's needs and I feel that the need to import Angular Material's CSS that are embedded through a component's ```styleUrls``` property simply adds bloat and maintenance burden we simply don't need.

Just so you have a perspective, my colleagues are consuming components such as Angular Material's ```MatFormField``` with ```matInput``` directives mainly for the purposes of using the functionality to do things such as showing validation errors, showing hint text of the number of characters remaining inside of a text field etc. However, ```MatFormField``` component injects the following CSS in its style urls:

Below fragment taken from https://github.com/angular/components/blob/master/src/material/form-field/form-field.ts

```
styleUrls: [
    'form-field.css',
    'form-field-fill.css',
    'form-field-input.css',
    'form-field-legacy.css',
    'form-field-outline.css',
    'form-field-standard.css',
  ]
```
The big kick in the gut is that because we have our own CSS that can take care of the look and feel of the some of these components but we are forced to firstly cancel out all of the CSS that is used in Angular material first before we can proceed to style our own thing. Furthermore, since we are not intending to use what is provided by Angular Material out of the box in terms of CSS styling, it's better that we might as well not include it all.

Furthermore, in terms of the form field component, you guys are not using any form of ```ViewEncapsulation``` at all, so apart from loading order of the CSS files, it doesn't make much of a difference to a global stylesheet, so I feel this request is not unreasonable.

I feel the people in my team are experienced enough in CSS that I feel we can take responsibility for our own actions of what happens there.

So basically we want to use your JS functionality but give us the option not to use your CSS and allow us the freedom and responsibility to maintain the look and feel ourselves. :)

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
I'm going to make the assumption of all versions of Angular Material that can be consumed in Angular 2 onwards since this is a conscious code structure decision made by the Angular Material team. All browsers.

**Comment by jelbourn** [comment]
I don't see this being something we ever do.
